src/fiatExchanges/PaymentMethodSection.tsx

```diff
@@ -0,0 +1,246 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Image, LayoutAnimation, StyleSheet, Text, View } from 'react-native'
+import { useSelector } from 'react-redux'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import Expandable from 'src/components/Expandable'
+import Touchable from 'src/components/Touchable'
+import {
+  CicoQuote,
+  getFeeValueFromQuotes,
+  PaymentMethod,
+  ProviderQuote,
+  SimplexQuote,
+} from 'src/fiatExchanges/utils'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+
+export interface PaymentMethodSectionProps {
+  paymentMethod: PaymentMethod
+  cicoQuotes: CicoQuote[]
+  setNoPaymentMethods: React.Dispatch<React.SetStateAction<boolean>>
+  quoteOnPress: (cicoQuote: CicoQuote) => void
+}
+
+export function PaymentMethodSection({
+  paymentMethod,
+  cicoQuotes,
+  setNoPaymentMethods,
+  quoteOnPress,
+}: PaymentMethodSectionProps) {
+  const { t } = useTranslation()
+  const sectionQuotes = cicoQuotes.filter(({ quote }) => quote.paymentMethod === paymentMethod)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+
+  const isExpandable = sectionQuotes.length > 1
+  const [expanded, setExpanded] = useState(false)
+
+  const toggleExpanded = () => {
+    LayoutAnimation.easeInEaseOut()
+    setExpanded(!expanded)
+  }
+  if (!sectionQuotes.length) {
+    setNoPaymentMethods(true)
+    return null
+  }
+
+  const renderExpandableSection = () => (
+    <>
+      <View style={styles.left}>
+        <Text style={styles.category}>
+          {paymentMethod === PaymentMethod.Card
+            ? t('selectProviderScreen.card')
+            : t('selectProviderScreen.bank')}
+        </Text>
+        {!expanded && (
+          <Text style={styles.fee}>
+            {renderFeeAmount(sectionQuotes[0].quote, t('selectProviderScreen.minFee'))}
+          </Text>
+        )}
+      </View>
+
+      <View style={styles.right}>
+        <Text style={styles.providerDropdown}>
+          {t('selectProviderScreen.numProviders', { count: sectionQuotes.length })}
+        </Text>
+      </View>
+    </>
+  )
+
+  const renderNonExpandableSection = () => (
+    <>
```

Comment:
nit: I might add error logging if sectionQuotes had length not equal to 1, since it violates an assumption this function relies on

Reply:
I don't think so. Thats one of those logs where its programmatically impossible to reach with the current code. Those types of issues should be covered by unit tests, which this is :)